plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -803,6 +859,83 @@ public Optional<ConnectorTableLayout> getInsertLayout(ConnectorSession session,
         return Optional.of(new ConnectorTableLayout(partitionColumnNames));
     }
 
+    @Override
+    public Iterator<RelationCommentMetadata> streamRelationComments(ConnectorSession session, Optional<String> schemaName, UnaryOperator<Set<SchemaTableName>> relationFilter)
+    {
+        List<RelationCommentMetadata> views = getViews(session, schemaName).entrySet().stream()
+                .map(entry -> RelationCommentMetadata.forRelation(entry.getKey(), entry.getValue().getComment()))
+                .toList();
+        Set<SchemaTableName> viewNames = views.stream().map(RelationCommentMetadata::name).collect(toUnmodifiableSet());
+
+        TrinoFileSystem fileSystem = fileSystemFactory.create(session);
+
+        List<RelationCommentMetadata> tables = listTables(session, schemaName).stream()
+                .filter(tableName -> !viewNames.contains(tableName))
+                .collect(collectingAndThen(toUnmodifiableSet(), relationFilter)).stream()
+                .map(tableName -> {
+                    if (redirectTable(session, tableName).isPresent()) {
+                        return RelationCommentMetadata.forRedirectedTable(tableName);
+                    }
+
+                    try {
+                        Optional<Table> metastoreTable = metastore.getRawMetastoreTable(tableName.getSchemaName(), tableName.getTableName());
+                        if (metastoreTable.isEmpty()) {
+                            // this may happen when table is being deleted concurrently,
+                            return null;
+                        }
+
+                        Table table = metastoreTable.get();
+                        verifyDeltaLakeTable(table);
+
+                        String tableLocation = HiveMetastoreBackedDeltaLakeMetastore.getTableLocation(table);
+                        if (cacheTableMetadata && table.getParameters().containsKey(TRINO_LAST_TRANSACTION_VERSION)) {
```

Comment:
Should we check if `TABLE_COMMENT ` similar to `TRINO_METADATA_SCHEMA_STRING)` check in `listTableColumns`?

Reply:
No, we don't store a table comment parameter in case of null. I will leave a code comment.